### ISSUE TITLE
IVFPQ recall decreases with increasing nprobe?

### ISSUE THREAD

**Issue by anirudhajith** [facebookresearch/faiss#2096]
# Summary

# Platform

OS: 

Faiss version:  faiss-gpu                 1.7.1           py3.8_h080d439_1_cuda10.2    pytorch

Installed from: Anaconda pip  

Running on:
- [x] CPU
- [ ] GPU

Interface: 
- [ ] C++
- [x] Python

# Reproduction instructions

I'm doing an experiment using custom (my own) centroids with the faiss IVFPQ index. I'm building an index using custom centroids using this function:
```python
def build_divfpq(flat_centroids, pq_centroids, index_embeddings):
    flatK, n = flat_centroids.shape
    D, K, _ = pq_centroids.shape

    print("Creating index...")
    quantizer = faiss.IndexFlatL2(n)
    index = faiss.IndexIVFPQ(quantizer, n, flatK, D, round(log2(K)))
    quantizer.add(flat_centroids)
    index.train(flat_centroids)
    faiss.copy_array_to_vector(pq_centroids.ravel(), index.pq.centroids)
    index.precompute_table()

    print("Indexing vectors...")
    index.add(index_embeddings)

    return index

```

I'm seeing that in some instances, the recall of a query is decreasing with increasing in nprobe. What does that mean? Isn't recall supposed to increase monotonically with increasing nprobe irrespective of what the centroids are?

```
Calculating recalls for NPROBE = 1...
recall@100: 0.1648

Calculating recalls for NPROBE = 10...
recall@100: 0.1495

Calculating recalls for NPROBE = 100...
recall@100: 0.1493
```

**Comment by mdouze** [comment]
absolutely, unless there is a misalignment between the metric used for the ground truth and the IVFPQ.
